src/main/java/com/aws/greengrass/shadowmanager/sync/model/FullShadowSyncRequest.java

```diff
@@ -193,6 +262,31 @@ && isDocVersionSame(localShadowDocument.get(), syncInformation, DataOwner.LOCAL)
                 .log("Successfully performed full sync");
     }
 
+    /**
+     * If this request was made as the result of merging, return
+     * a list of all the "sub-requests" that require execution,
+     * as deemed by {@link SyncRequest#isUpdateNecessary(SyncContext)}.
+     *
+     * @param context sync context
+     * @return sync requests
+     * @throws RetryableException       When error occurs in sync operation indicating a request needs to be retried
+     * @throws SkipSyncRequestException When error occurs in sync operation indicating a request needs to be skipped.
+     * @throws UnknownShadowException   When shadow not found in the sync table.
+     */
+    private List<SyncRequest> getNecessaryMergedRequests(SyncContext context)
+            throws RetryableException, UnknownShadowException, SkipSyncRequestException {
+        if (mergedRequests == null) {
+            return null;
+        }
+        List<SyncRequest> necessaryUpdates = new ArrayList<>();
+        for (SyncRequest request : mergedRequests) {
+            if (request.isUpdateNecessary(context)) {
+                necessaryUpdates.add(request);
+            }
+        }
+        return necessaryUpdates;
```

Comment:
nit: you could do something like:
```
List<SyncRequest> necessaryUpdates = mergedRequests.stream().filter(r -> r.isUpdateNecessary(context)).collect(Collectors.toList());

Reply:
is update necessary throws, so would have to wrap it in a runtime exception which i don't think we want